src/logic/calculate.js

```diff
@@ -0,0 +1,40 @@
+import operate from './operate';
+
+const calculate = (data, buttonName) => {
+  let { total, next, operation } = { ...data };
```

Comment:
I don't think you need a spread operator in this case.
simple `  let { total, next, operation } = data;` would be fine